packages/contracts/contracts/LenderCommitmentForwarder.sol

```diff
@@ -192,25 +339,105 @@ contract LenderCommitmentForwarder is TellerV2MarketForwarder {
             "Commitment has expired"
         );
 
+        require(
+            commitment.maxPrincipalPerCollateralAmount == 0 || 
+                _collateralAmount >= getRequiredCollateral(_principalAmount, commitment.maxPrincipalPerCollateralAmount),
+
+            "Insufficient collateral"
+        );
+
+        if(commitment.collateralTokenType == CommitmentCollateralType.ERC721 ||
+        commitment.collateralTokenType == CommitmentCollateralType.ERC721_ANY_ID ){
+              require( _collateralAmount == 1, "Invalid amount for ERC721");
+        }
+
+        if(commitment.collateralTokenType == CommitmentCollateralType.ERC721 ||
+        commitment.collateralTokenType == CommitmentCollateralType.ERC1155 ){
+              require( commitment.collateralTokenId == _collateralTokenId, "Invalid tokenId");
+        }
+      
+
+        bidId = _submitBidFromCommitment(
+            borrower,
+            _marketId,
+            commitment.principalTokenAddress,
+            _principalAmount,
+            commitment.collateralTokenAddress,
+            _collateralAmount,
+            _collateralTokenId,
+            commitment.collateralTokenType,
+            _loanDuration,
+            _interestRate
+        );
+
+        _acceptBid(bidId, commitment.lender);
+
+        _decrementCommitment(_commitmentId, _principalAmount);
+
+        emit ExercisedCommitment(
+            _commitmentId,
+            borrower,
+            _marketId,
+            commitment.principalTokenAddress,
+            _principalAmount,
+            bidId,
+            commitment.collateralTokenAddress,
+            _collateralAmount,
+            _collateralTokenId
+        );
+    }
+
+    function getRequiredCollateral(uint256 _principalAmount, uint256 _maxPrincipalPerCollateralAmount) public view virtual returns (uint256 _collateralAmountRaw) {
+        _collateralAmountRaw= (_principalAmount * _maxPrincipalPerCollateralAmount) /  PRINCIPAL_PER_COLLATERAL_EXPANSION_FACTOR  ;
```

Comment:
I don't think this is correct - I believe you want to swap the `_maxPrincipalPerCollateralAmount` and `PRNCIPAL_PER_COLLATERAL_EXPANSION_FACTOR` variables..

Note: assume values below are normalized (decimals aren't being shifted)

| variable | value 
-: | :-
| principal | 800
| max principal per collateral | 500 |
| expected collateral | 1.6 (`800 / 500`) |

actual result:
```
800 * 500 = 400000
```

Reply:
also in this example the required collateral amount `1.4` won't work for NFTs so we would would need to round this number up to the next whole number 

you can use OZ's `Math.mulDiv(x, y, denominator, Rounding.Up)` function
https://github.com/OpenZeppelin/openzeppelin-contracts/blob/6d18435098c797bc8c3cfbf51fe10d450a2f6595/contracts/utils/math/Math.sol#L133-L136

Reply:
done